### DISCUSSION TITLE
auth_type in HttpSensor

### DISCUSSION THREAD

**Discussion by sudarshan2906** [apache/airflow#43118] [General]
HttpOperator has an option auth_type which enables us to add a custom auth class. 

Using this we can use APIs which has OAUTH authentication as mentioned here in this article: https://medium.com/@andrewhharmon/airflow-httpoperator-with-oauth-c5e7c8fa2232

I don't see this option added in HttpSensor, Is it possible to use HttpSensor for APIs which has OAUTH as authentication?

**Comment by potiuk** [comment]
If it's not there - then likely not - but as usual in Airflow, you can look at the code of Http Operator and implement the same thing in your custom HttpSensor ([following the docs](https://airflow.apache.org/docs/apache-airflow/stable/howto/custom-operator.html#creating-a-custom-operator)) and then even contribute it back to Airflow code.
